File: jadx/src/main/java/com/mcal/jadx/Smali2Java.java
package com.mcal.jadx;

import java.io.File;

import jadx.api.JadxArgs;
import jadx.api.JadxDecompiler;
import jadx.api.JavaClass;

public class Smali2Java {
    public static boolean decompile(String errorMessage, String dexPath, String outputDir) {
        try {
            File dexInputFile = new File(dexPath);
            File javaOutputDir = new File(outputDir);

            JadxArgs args = new JadxArgs();
            args.setOutDirSrc(javaOutputDir);
            args.setSkipResources(true);
            args.setShowInconsistentCode(true);
            args.setInputFile(dexInputFile);

            JadxDecompiler decompiler = new JadxDecompiler(args);
            decompiler.load();
            decompiler.saveSources();

            JavaClass javaClass = decompiler.getClasses().iterator().next();
            javaClass.decompile();
            return true;
        } catch (Exception | StackOverflowError e) {
            errorMessage = "Cannot decompile java code: " + e.getMessage();
        }
        return false;
    }
}

File: jadx/src/main/java/com/mcal/jadx/Smali2Java.kt
package com.mcal.jadx

import jadx.api.JadxArgs
import jadx.api.JadxDecompiler
import java.io.File
import java.lang.Exception

object Smali2Java {
    @JvmStatic
    fun decompile(errorMessage: String?, dexPath: String?, outputDir: String?): Boolean {
        var errorMessage = errorMessage
        try {
            val dexInputFile = File(dexPath)
            val javaOutputDir = File(outputDir)
            val args = JadxArgs()
            args.outDirSrc = javaOutputDir
            args.isSkipResources = true
            args.isShowInconsistentCode = true
            args.setInputFile(dexInputFile)
            val decompiler = JadxDecompiler(args)
            decompiler.load()
            decompiler.saveSources()
            val javaClass = decompiler.classes.iterator().next()
            javaClass.decompile()
            //javaClass.getCode();
            return true
        } catch (e: Exception) {
            errorMessage = "Cannot decompile java code: " + e.message
            e.printStackTrace()
        } catch (e: StackOverflowError) {
            errorMessage = "Cannot decompile java code: " + e.message
            e.printStackTrace()
        }
        return false
    }
}